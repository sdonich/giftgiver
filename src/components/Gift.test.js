import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';


import Gift from './Gift';

describe('Gift', () => {
  const gift = shallow(<Gift />);

  test('renders properly', () => {
    const tree = renderer.create(<Gift />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('initializes a person and present in `state`', () => {
    expect(gift.state()).toEqual({ person: '', present: '' });
  });

  describe('when typing into the person input', () => {
    const person = 'Uncle'
    beforeEach(() => {
      gift.find('.input-person').simulate('change', { target: { value: person } });
    })
    test('updates the person in `state`', () => {
      expect(gift.state().person).toBe(person);
    });
  });
  
  describe('when typinng into the present input', () => {
    const present = 'Golf Clubs';
    beforeEach(() => {
      gift.find('.input-present').simulate('change', {target: {value: present}});
    });
    test('updates the present in `state`', () => {
      expect(gift.state().present).toBe(present);
    });
  });
});