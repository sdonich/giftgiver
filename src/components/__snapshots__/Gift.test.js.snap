// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gift renders properly 1`] = `
<div>
  <form
    className=""
  >
    <div
      className="form-group"
    >
      <label
        className="form-label"
      >
        Person
      </label>
      <input
        className="input-person form-control"
        onChange={[Function]}
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="form-label"
      >
        Present
      </label>
      <input
        className="input-present form-control"
        onChange={[Function]}
      />
    </div>
  </form>
</div>
`;
